resources/js/common/components/PlayerBadgeIndicator/PlayerBadgeIndicator.tsx

```diff
@@ -0,0 +1,30 @@
+import type { FC } from 'react';
+
+import { AwardType } from '@/common/utils/generatedAppConstants';
+import { getLabelFromPlayerBadge } from '@/common/utils/getLabelFromPlayerBadge';
+import { cn } from '@/utils/cn';
+
+type PlayerBadgeIndicatorProps = App.Platform.Data.PlayerBadge & { className?: string };
+
+export const PlayerBadgeIndicator: FC<PlayerBadgeIndicatorProps> = ({
+  awardType,
+  awardDataExtra,
+  className,
+}) => {
+  return (
+    <div
+      aria-label={`${getLabelFromPlayerBadge(awardType, awardDataExtra)} indicator`}
+      className={cn(
+        'h-2 w-2 rounded-full',
+
+        awardType === AwardType.Mastery && awardDataExtra ? 'bg-[gold] light:bg-yellow-600' : '', // Mastered
+        awardType === AwardType.Mastery && !awardDataExtra ? 'border border-yellow-600' : '', // Completed
+
+        awardType === AwardType.GameBeaten && awardDataExtra ? 'bg-zinc-300' : '', // Beaten
+        awardType === AwardType.GameBeaten && !awardDataExtra ? 'border border-zinc-400' : '', // Beaten (softcore)
+
+        className,
```

Comment:
Yep.

There are a few problems with the current indicators that I'm trying to solve with this new design. The current indicators:
* Are quite loud when "stacked" in multiple successive rows.
* Can visually conceal progress near the end of the bar.
* Have labels which are impossible to see without a mouse to hover and read the `title` attribute.

The new indicators solve all of these problems. On hover, tap, or focus, the label appears. They're also slightly more visually subdued when "stacked":

![Screenshot 2024-09-23 at 4 11 09 PM](https://github.com/user-attachments/assets/3ad9f141-daa4-4df3-9f9f-4993bf3d5776)

Reply:
What about something more similar to the Completion Progress page?
![image](https://github.com/user-attachments/assets/0ef442d7-c4ac-49a4-81f6-55a034e4db71)

or the Completion Progress on the profile?
![image](https://github.com/user-attachments/assets/ad380434-5b1e-4f8b-843d-a42d2f0d19be)

Both have text indicating how full the bar is.

The way you have it now, it's just too similar to the bar itself. It feels like there's a baby second progress bar and I just can reason why it would be there.

Maybe it would be less off-putting if the Mastered text was always visible instead of just appearing on mouseover.
![image](https://github.com/user-attachments/assets/56d1c7e1-567c-47d5-9abf-b0ba66339a5b)

Reply:
I'm doing my best to mitigate a disorienting sensation I experience when I see this:

![Screenshot 2024-09-25 at 5 45 31 PM](https://github.com/user-attachments/assets/503abc04-0ddb-4bfb-b253-d08dee941077)

My eyes feel like they're uncomfortably glued to these labels, drawing my attention away from nearly everything else on the page. The other pages' bars either do something similar when there's a lot of them, or would cause other undesirable effects on the rows of the grid, such as height or padding inflation.

The rows on this table are slim enough that I even feel this sensation when just percentages are shown under a lot of these bars within close proximity.

Here's where I've landed in latest which should hopefully be a healthy compromise:


https://github.com/user-attachments/assets/884c12f5-f616-45c2-9967-a5b3ee8d4986

Reply:
What do you think about left-aligning the second line?
![image](https://github.com/user-attachments/assets/aa69ced9-6af8-48c3-8201-9c3570c0dffd)

Reply:
I like that idea. Updated in latest.